pkgs/by-name/ka/kana/package.nix

```diff
@@ -0,0 +1,62 @@
+{ lib
+, stdenv
+, fetchFromGitLab
+, rustPlatform
+, meson
+, ninja
+, pkg-config
+, rustc
+, cargo
+, wrapGAppsHook4
+, desktop-file-utils
+, libadwaita
+, gst_all_1
+}:
+
+stdenv.mkDerivation rec {
+  pname = "kana";
+  version = "1.4";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "fkinoshita";
+    repo = "Kana";
+    rev = "v${version}";
```

Comment:
```suggestion
    rev = "v${finalAttrs.version}";
```

Reply:
what is the actual advantage of this? it's not like overriding `version` will work properly, since all the fetchers are fixed-output

in fact, this could even cause a derivation to build fine on one system, but fail on another (i've had this happen when updating packages and forgetting to invalidate the source hash, this could be extremely confusing to less technical users who haven't touched the inside of nixpkgs)

Reply:
> what is the actual advantage of this?

[Get rid of `rec`](https://nix.dev/guides/best-practices.html#recursive-attribute-set-rec).

Reply:
but the only problem with `rec` that page mentions is the possibility for infinite recursion, which is equally possible with the finalAttrs pattern.

i asked what the advantage of getting rid of `rec` is, and you gave me quite a recursive answer.